libs/core/src/lib/split-button/split-button.component.ts

```diff
@@ -55,8 +58,15 @@ export class SplitButtonComponent {
     disabled: boolean;
 
     /** The Title for main  action button */
-    @Input()
-    mainActionTitle: string;
+    private _mainActionTitle: string;
+
+    set mainActionTitle(title: string) {
+        this._mainActionTitle = title;
+    }
+
+    get mainActionTitle(): string {
+        return this._mainActionTitle;
+    }
```

Comment:
I think we should keep mainActionTitle, just in case user wants to define his own. If it's not set, then it should be changed. 
Maybe you can put some flag eg `@Input() changeTitleOnSelection: boolean`, which will define if selection of item should change action/title

Reply:
Well if he uses template example he can set the title. Look at the split button template example.

Reply:
Actually what i did is i made by default it add the option one, if the user adds a mainItemTitle it will use that instead